### ISSUE TITLE
Options for no-empty rule

### ISSUE THREAD

**Issue by cyrilletuzi** [palantir/tslint#2970]
### Bug Report

- __TSLint version__: last
- __TypeScript version__: last
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
try {
  JSON.parse('');
} catch (error) {}

fetch('/api').then((res) => res.json()).catch(() => {});
```

with `tslint.json` configuration:

```json
{ "no-empty": true }
```

#### Actual behavior

Error raised as they are empty blocks.

#### Expected behavior

The `no-empty` rule is great because it avoids to keep unnecessary code.

But it's too generic for today TS development : in the examples above, it's important to catch errors, but we don't always have a real action to do in case of error (which is not the same as not catching the error at all). So here, empty blocks are right.

Same as #2107, but as the first example shows, it is not limited to functions. So different options would be nice, to avoid unnecessary conditions, loops, etc., but allow empty callbacks or empty try/catch.
